kubetest/main.go

```diff
@@ -715,10 +715,15 @@ func chmodArtifacts() error {
 }
 
 func prepare(o *options) error {
+	k8sProviderEnvVar := o.provider
+	if o.provider == "kubernetes-anywhere" {
+		// To make gce-related setup scripts in k8s to get sourced.
+		k8sProviderEnvVar = "gce"
```

Comment:
Agreed, this sounds wrong. This is very likely going to trigger all sorts of behavior that reaches around kubernetes-anywhere and attempts to directly interact with GCE instead. Our tests currently all happen to use GCE, but kubernetes-anywhere has support for more cloud providers than that, and we know we want to add more e2e test jobs that use kubernetes-anywhere with other providers than GCE.